dhis-2/dhis-api/src/main/java/org/hisp/dhis/sms/config/GenericGatewayParameter.java

```diff
@@ -106,4 +107,30 @@ public void setEncode( boolean encode )
     {
         this.encode = encode;
     }
+
+    @Override
+    public boolean equals( Object o )
```

Comment:
Can't we just annotate this class with:
```java
@EqualsAndHashCode(of = {"key","value", "confidential", "encode", "header"})
```
?

Reply:
resolved